slot.go

```diff
@@ -134,30 +176,75 @@ func (s *Slot) findKey(search []*pkcs11.Attribute) (*Key, error) {
 func describeKey(obj obj) (c, t string, err error) {
 	attrs, err := obj.ctx.GetAttributeValue(obj.h, obj.o, []*pkcs11.Attribute{
 		&pkcs11.Attribute{Type: pkcs11.CKA_CLASS},
-		&pkcs11.Attribute{Type: pkcs11.CKA_KEY_TYPE},
 	})
 	if err != nil {
+		err = fmt.Errorf("Error reading CKA_CLASS: %x", err)
```

Comment:
Errors are typically not capitalized (so they compose better), and commonly in libraries they also include the name of the package that generated them. It's also generally clear that error values are errors, so it's often okay to omit the string `error` from error messages. This one might read `while reading CKA_CLASS: %v` (not `%x`, which is hex)